### ISSUE TITLE
CMake uses C++-specific options for C files too

### ISSUE THREAD

**Issue by vadz** [wxWidgets/wxWidgets#22696]
Doing

    $ cmake -S . -B ~/build/wx/cmake -GNinja
    $ cmake --build ~/build/wx/cmake

results in the following output
```
[2/794] Building C object libs/base/CMakeFiles/wxbase.dir/__/__/__/__/src/common/extended.c.o
cc1: warning: command-line option ‘-Wno-ctor-dtor-privacy’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-Woverloaded-virtual’ is valid for C++/ObjC++ but not for C
[5/794] Building C object libs/core/CMakeFiles/wxcore.dir/__/__/__/__/src/gtk/treeentry_gtk.c.o
cc1: warning: command-line option ‘-Wno-ctor-dtor-privacy’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-Woverloaded-virtual’ is valid for C++/ObjC++ but not for C
```

It's harmless, but it would be nice if we could avoid these warnings by using separate C and C++ warning options.

**Comment by MaartenBent** [comment]
This can be fixed by adding those warnings only for CXX compiler, like: `$<$<COMPILE_LANGUAGE:CXX>:${cpp_only_warnings}>`

I'll create a PR with this.

**Comment by vadz** [comment]
I think this should be safe to backport to 3.2, shouldn't it?

**Comment by MaartenBent** [comment]
Yes, this can be backported.

**Comment by vadz** [comment]
Thanks, I've actually already done it and forgot to close this (it's really inconvenient not to be able to use `Closes #NNNN` in 3.2 branch).
